airbyte-integrations/connectors/destination-rockset/src/main/java/io/airbyte/integrations/destination/rockset/RocksetWriteApiConsumer.java

```diff
@@ -0,0 +1,149 @@
+
+import static io.airbyte.integrations.destination.rockset.RocksetUtils.*;
+
+import com.fasterxml.jackson.core.type.TypeReference;
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.rockset.client.ApiClient;
+import com.rockset.client.api.DocumentsApi;
+import com.rockset.client.model.AddDocumentsRequest;
+import io.airbyte.integrations.base.AirbyteMessageConsumer;
+import io.airbyte.protocol.models.AirbyteMessage;
+import io.airbyte.protocol.models.ConfiguredAirbyteCatalog;
+import io.airbyte.protocol.models.DestinationSyncMode;
+
+import java.time.Duration;
+import java.util.concurrent.CompletableFuture;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+import java.util.concurrent.TimeUnit;
+import java.util.function.Consumer;
+import java.util.stream.Collectors;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class RocksetWriteApiConsumer implements AirbyteMessageConsumer {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(RocksetWriteApiConsumer.class);
+    private static final ObjectMapper mapper = new ObjectMapper();
+    // IO bound tasks, use cached thread pool
+    private static final ExecutorService exec = Executors.newCachedThreadPool();
+
+    private final String apiKey;
+    private final String apiServer;
+    private final String workspace;
+
+    private final ConfiguredAirbyteCatalog catalog;
+    private final Consumer<AirbyteMessage> outputRecordCollector;
+
+    private final RocksetSQLNameTransformer nameTransformer = new RocksetSQLNameTransformer();
+
+    private ApiClient client;
+
+    public RocksetWriteApiConsumer(
+            JsonNode config,
+            ConfiguredAirbyteCatalog catalog,
+            Consumer<AirbyteMessage> outputRecordCollector) {
+        this.apiKey = config.get(API_KEY_ID).asText();
+        this.apiServer = config.get(API_SERVER_ID).asText();
+        this.workspace = config.get(WORKSPACE_ID).asText();
+
+        this.catalog = catalog;
+        this.outputRecordCollector = outputRecordCollector;
+    }
+
+    @Override
+    public void start() throws Exception {
+        this.client = RocksetUtils.apiClient(apiKey, apiServer);
+        LOGGER.info("Creating workspace");
+
+        RocksetUtils.createWorkspaceIfNotExists(client, workspace);
+
+        CompletableFuture<?>[] overwrittenStreams = catalog.getStreams()
+                .stream()
+                .filter(s -> s.getDestinationSyncMode() == DestinationSyncMode.OVERWRITE)
+                .map(s -> s.getStream().getName())
+                .map(nameTransformer::convertStreamName)
+                .map(this::emptyCollection)
+                .collect(Collectors.toList())
+                .toArray(CompletableFuture[]::new);
+
+        CompletableFuture<?>[] appendStreams = catalog.getStreams().stream()
+                .filter(s -> s.getDestinationSyncMode() != DestinationSyncMode.OVERWRITE)
```

Comment:
This has been updated as 
`.filter(s -> s.getDestinationSyncMode() == DestinationSyncMode.APPEND)`
because dedupe mode isn't supported **yet** and append streams will not delete any old documents as would be done in overwrite mode.